prober/tcp.go

```diff
@@ -158,6 +158,23 @@ func ProbeTCP(ctx context.Context, target string, module config.Module, registry
 			}
 			probeFailedDueToRegex.Set(0)
 			send = string(qr.Expect.Regexp.Expand(nil, []byte(send), scanner.Bytes(), match))
+			if qr.Labels != nil {
```

Comment:
can you take this logic out into a function and call that in the ProbeTCP?

please add some tests for this logic (maybe one in ProbeTCP test and then add some unit tests for the extracted func.)

Reply:
Refactored. I hope I used references correctly.
(I was good at C++ some 20 years ago, but this is basically my first Go `func`)